app/models/concerns/reviews/score.rb

```diff
@@ -106,6 +110,40 @@ def score_by_splitted_effectiveness date
                      end
   end
 
+  def score_by_weighted_weaknesses date
+    weaknesses      = has_final_review? ? final_weaknesses : self.weaknesses
+    total           = 0
+    high_score      = 150
+    medium_score    = 50
+    hundred_percent = 100
+
+    scores = weaknesses.select { |w| w.state_weight > 0 }.group_by do |w|
+      [w.risk_weight, w.state_weight, w.age_weight(date: date)]
+    end
+
+    scores.each do |row, weakness|
```

Comment:
```suggestion
    total = scores.sum do |row, weakness|
```
Y el resto te lo dejo de tarea.